zproject/backends.py

```diff
@@ -736,12 +736,22 @@ def social_associate_user_helper(backend: BaseAuth, return_data: Dict[str, Any],
     return_data['validated_email'] = validated_email
     user_profile = common_get_active_user(validated_email, realm, return_data)
 
-    if 'fullname' in kwargs["details"]:
+    if 'fullname' not in kwargs["details"]:
```

Comment:
It might be better to use more variables here, e.g. `full_name = kwargs['details'].get('fullname')`, and then` if full_name is None:` just to avoid repeating parsing code.